File: app/src/main/java/com/sc/clgg/activity/basic/MainActivity.java
//package com.sc.clgg.activity.basic;
//
//import android.content.Intent;
//import android.os.Bundle;
//import android.support.annotation.Nullable;
//import android.support.v4.app.Fragment;
//import android.suppLort.v7.app.AppCompatActivity;
//import android.text.TextUtils;
//import android.view.KeyEvent;
//import android.view.View;
//import android.widget.TextView;
//
//import com.sc.clgg.R;
//import com.sc.clgg.activity.fragment.TransportManageFragment;
//import com.sc.clgg.activity.fragment.UserSettingsFragment;
//import com.sc.clgg.activity.fragment.VehicleManageFragment;
//import com.sc.clgg.adapter.FragmentAdapter;
//import com.sc.clgg.bean.MessageEvent;
//import com.sc.clgg.dialog.ExitDialog;
//import com.sc.clgg.dialog.SetGpsDialog;
//import com.sc.clgg.util.ConfigUtil;
//import com.sc.clgg.widget.NotSlideViewPager;
//
//import org.greenrobot.eventbus.EventBus;
//import org.greenrobot.eventbus.Subscribe;
//import org.greenrobot.eventbus.ThreadMode;
//
//import java.util.ArrayList;
//import java.util.List;
//
//import butterknife.BindView;
//import butterknife.BindViews;
//import butterknife.ButterKnife;
//import butterknife.OnClick;
//import toolbox.helper.AppHelper;
//
//
///**
// * 创建时间：2017/7/31 11:20
// *
// * @author lvke
// */
//public class MainActivity extends AppCompatActivity {
//    @BindView(R.id.viewpager)
//    NotSlideViewPager mViewPager;
//
//    @BindView(R.id.tv_unread)
//    TextView mUnReadTv;
//
//    @BindViews({R.id.tv_truck_manage, R.id.tv_transport_manage, R.id.tv_user_setttings})
//    List<TextView> textViews;
//
//    private SetGpsDialog setGpsDialog;
//    /**
//     * 标记当前底部的index
//     */
//    private int currenMainTabIndex;
//
//    @Override
//    protected void onCreate(@Nullable Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
//        ButterKnife.bind(this);
//
//        List<Fragment> fragmentList = new ArrayList<>();
//        fragmentList.add(new VehicleManageFragment());
//        fragmentList.add(new TransportManageFragment());
//        fragmentList.add(new UserSettingsFragment());
//        FragmentAdapter fragmentAdapter = new FragmentAdapter(getSupportFragmentManager(), fragmentList);
//
//        mViewPager.setOffscreenPageLimit(2);
//        mViewPager.setAdapter(fragmentAdapter);
//
//        setGpsDialog = new SetGpsDialog(this);
//    }
//
//    @Override
//    protected void onResume() {
//        super.onResume();
//        if (!AppHelper.isGpsOpen(getApplicationContext()) && !setGpsDialog.getDialog().isShowing()) {
//            setGpsDialog.show();
//        }
//    }
//
//    private void showTab(int showIndex, List<TextView> textViews) {
//        for (TextView tv : textViews) {
//            if (textViews.indexOf(tv) == showIndex) {
//                tv.setSelected(true);
//            } else {
//                tv.setSelected(false);
//            }
//        }
//    }
//
//    public int getCurrenMainTabIndex() {
//        return currenMainTabIndex;
//    }
//
//    public void setCurrenMainTabIndex(int currenMainTabIndex) {
//        this.currenMainTabIndex = currenMainTabIndex;
//    }
//
//    @OnClick({R.id.tv_truck_manage, R.id.tv_transport_manage, R.id.tv_user_setttings})
//    public void a(View view) {
//        if (view.getId() == R.id.tv_truck_manage) {
//            mViewPager.setCurrentItem(0, false);
//            setCurrenMainTabIndex(0);
//            showTab(0, textViews);
//        } else if (view.getId() == R.id.tv_transport_manage) {
//            mViewPager.setCurrentItem(1, false);
//            setCurrenMainTabIndex(1);
//            showTab(1, textViews);
//        } else if (view.getId() == R.id.tv_user_setttings) {
//            if (TextUtils.isEmpty(new ConfigUtil().getUserid())) {
//                startActivity(new Intent(this, LoginActivity.class));
//            } else {
//                mViewPager.setCurrentItem(2, false);
//                setCurrenMainTabIndex(2);
//                showTab(2, textViews);
//            }
//        }
//    }
//
//    @Override
//    protected void onStart() {
//        super.onStart();
//        EventBus.getDefault().register(this);
//    }
//
//    @Override
//    protected void onStop() {
//        EventBus.getDefault().unregister(this);
//        super.onStop();
//    }
//
//    @Subscribe(threadMode = ThreadMode.MAIN)
//    public void onMessageEvent(MessageEvent event) {
//        int messageNumber = 99;
//        if (event.getValue() > messageNumber) {
//            mUnReadTv.setText(R.string.more_than_99);
//        } else {
//            mUnReadTv.setText(String.valueOf(event.getValue()));
//        }
//    }
//
//    @Override
//    public boolean onKeyDown(int keyCode, KeyEvent event) {
//        if (keyCode == KeyEvent.KEYCODE_BACK) {
//
//            new ExitDialog(this).show("退出提示", "确定退出车轮滚滚？", (arg0, arg1) -> System.exit(0));
//
//            return true;
//        }
//        return super.onKeyDown(keyCode, event);
//    }
//}

File: app/src/main/java/com/sc/clgg/activity/basic/MainActivity.kt
package com.sc.clgg.activity.basic

import android.os.Bundle
import android.view.KeyEvent
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.gyf.barlibrary.ImmersionBar
import com.sc.clgg.R
import com.sc.clgg.activity.fragment.*
import com.sc.clgg.adapter.FragmentAdapter
import com.sc.clgg.bean.Banner
import com.sc.clgg.bean.MessageEvent
import com.sc.clgg.dialog.ExitDialog
import kotlinx.android.synthetic.main.activity_main.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode


/**
 * @author：lvke
 * @date：2018/2/27 11:08
 */

class MainActivity : AppCompatActivity() {
    private lateinit var textViews: List<TextView>

    /**
     * 标记当前底部的index
     */
    private var currenMainTabIndex: Int = 0

    companion object {
        var truckFriendsFragment: TruckFriendsFragment? = null

        var mallFragment: MallFragment? = null
    }

    var bannerData: ArrayList<Banner.Bean>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewpager?.offscreenPageLimit = 5
        truckFriendsFragment = TruckFriendsFragment()
        mallFragment = MallFragment()
        viewpager?.adapter = FragmentAdapter(supportFragmentManager, listOf(HomeFragment(), CarNetFragment(), mallFragment, truckFriendsFragment, MyFragment()))

        textViews = listOf(tv_home, tv_car_net, tv_mall, tv_truck_circle, tv_my)

        tv_home.setOnClickListener { checked(0) }
        tv_car_net.setOnClickListener { checked(1) }
        tv_mall.setOnClickListener { checked(2) }
        tv_truck_circle.setOnClickListener { checked(3) }
        tv_my.setOnClickListener { checked(4) }
        ImmersionBar.with(this).init()

        checked(0)
        tv_home.isSelected = true
    }

    override fun onDestroy() {
        super.onDestroy()
        ImmersionBar.with(this).destroy()
    }

    private fun showTab(showIndex: Int, textViews: List<TextView>) {
        for (tv in textViews) tv.isSelected = textViews.indexOf(tv) == showIndex
    }

    fun checked(i: Int) {
        viewpager?.setCurrentItem(i, false)
        currenMainTabIndex = i
        showTab(i, textViews)
    }

    override fun onStart() {
        super.onStart()
        EventBus.getDefault().register(this)
    }

    override fun onStop() {
        EventBus.getDefault().unregister(this)
        super.onStop()
    }

    @Subscribe(threadMode = ThreadMode.MAIN, sticky = true)
    fun onMessageEvent(event: MessageEvent) {
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            if (currenMainTabIndex == 2 && mallFragment?.canGoBack()!!) {
                mallFragment?.goBack()
            } else {
                ExitDialog(this).show("退出提示", "确定退出车轮滚滚？") { _, _ -> System.exit(0) }
            }
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

}
